Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Spawning/DespawnNodeable.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/TickBus.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+#include <ScriptCanvas/CodeGen/NodeableCodegen.h>
+#include <ScriptCanvas/Core/Node.h>
+#include <ScriptCanvas/Core/Nodeable.h>
+#include <ScriptCanvas/Libraries/Spawning/SpawnTicketInstance.h>
+#include <Include/ScriptCanvas/Libraries/Spawning/DespawnNodeable.generated.h>
+
+namespace ScriptCanvas::Nodeables::Spawning
+{
+    //! Node for despawning entities
+    class DespawnNodeable
+        : public ScriptCanvas::Nodeable
+        , public AZ::TickBus::Handler
+    {
+        SCRIPTCANVAS_NODE(DespawnNodeable);
+    public:
+        DespawnNodeable() = default;
+        DespawnNodeable(const DespawnNodeable& rhs);
+        DespawnNodeable& operator=(DespawnNodeable& rhs);
+
+        // ScriptCanvas::Nodeable  overrides ...
+        void OnInitializeExecutionState() override;
+        void OnDeactivate() override;
+
+        // AZ::TickBus::Handler overrides ...
+        void OnTick(float delta, AZ::ScriptTimePoint timePoint) override;
+        
+    private:
+        AZStd::vector<SpawnTicketInstance> m_despawnedTicketList;
```

Comment:
I'm not sure if this needs to be a vector of tickets. Same with the vector in SpawnNodeable. We will either spawn all entities or despawn all entities corresponding to 1 ticket. So I can't see a scenario where one node can result in multiple tickets. Unless I'm missing something here...

Reply:
So this was mostly copied from original SpawnNodeable. I've only simplified it and replaced ticketId with actual ticket instance to keep it in memory until Despawn callback is called. It's a synchronization mechanism. Despawn callback will be (eventually) executed on a different thread, and so there's no guarantee it will be called before another request to despawn something else arrives. That is why all tickets are aggregated in a collection, and processed in OnTick function. Here's the callback function:
```
auto despawnCompleteCB = [this, spawnTicket]([[maybe_unused]] AzFramework::EntitySpawnTicket::Id ticketId)
        {
            AZStd::lock_guard lock(m_mutex);
            // SpawnTicket instance is cached instead of SpawnTicketId to simplify managing its lifecycle on Script Canvas
            // and to provide easier access to it in OnDespawn callback
            m_despawnedTicketList.push_back(spawnTicket);
        };
```

Despawn node can be called multiple times:
![image](https://user-images.githubusercontent.com/82239319/155819531-1eb5f5b7-34f3-4e11-ac3e-a7d969f184c3.png)
we just need to pass SpawnTicket instance as a parameter. We can have a single Despawn node to despawn everything in the game.